qmcpy/algorithms/distribution/__init__.py

```diff
@@ -51,13 +51,13 @@ def __setitem__(self,i,val): self.distrib_list[i]=val
     def __repr__(self): return univ_repr(self,'distrib_list')
 
 
-class Measure():
+class Measure(object):
```

Comment:
The standard is to always inherit from something (the base being `object`).  It's not technically required, but not doing so is frowned upon.

Reply:
Measure is now an abstract class (inherits from ABC) with similar construction to DiscreteDistribution

Reply:
Is there some functionality in `ABC` that you are using here.  I didn't immediately see anything that actually required it.  I would, in general, avoid using `ABC` unless there's a specific functionality that's required (which can happen but is generally not the case in Python).

Reply:
Is there some functionality in ABC that you are using here. I didn't immediately see anything that actually required it. I would, in general, avoid using ABC unless there's a specific functionality that's required (which can happen but is generally not the case in Python).
@mikemccourt

Reply:
Also, sorry, I accidentally posted something on here on my SigOpt account -- I'll need to be more careful about that.

Reply:
It was changed to an abstract (abc) class to allow for behind-the-scenes multi-dimensional measure construction (similar to AsianCallOption and the Abstract class DiscreteDistribution)
Just makes it easier for contributes to add their own measures without having to worry about creating measure objects that act as lists of measure objects

Reply:
I don't think that's actually true -- I think if you just inherit from `object` and `raise NotImplementedError` in undefined functions it will do the same thing (without the more complicated abstract structure in place).

In my mind, the only 2 cases where `ABC` is actually useful are

* in circumstances where instantiation of a class is extremely expensive and you want to throw an error at instantiation time if someone tries to instantiate a pure virtual class (as opposed to simply erroring when it gets used), or
* if you are leveraging duck typing throughout the code and want to replace `hasattr` calls with `isinstance` or `issubclass` calls.

Beyond that, I don't think there's too much benefit to using `ABC`.  Again, I'm not going to say what you guys need to do, but I do think that the use of `ABC` is pretty uncommon unless a programmer coming from a Java background.